### ISSUE TITLE
Update class DataLoaderIterWrapper to accommodate extra returned values

### ISSUE THREAD

**Issue by vpapaioannou** [davidtvs/pytorch-lr-finder#18]
At https://github.com/davidtvs/pytorch-lr-finder/blob/master/torch_lr_finder/lr_finder.py#L453

` inputs, labels = next(self._iterator)`

Could it be changed to

` inputs, labels, *rest = next(self._iterator)`

to accommodate cases where more values are returned?

For example, if a weighted loss is used, then also the weights need to be returned to calculate the loss. This weighted loss can be found in U-Net original paper. Another example is to return the training data file name being used; this is useful for debugging.

**Comment by davidtvs** [comment]
Is the suggested change only about not getting an error when  `next(self._iterator)` returns more than values or do you expect `*rest` to be used in some way during training?

**Comment by vpapaioannou** [comment]
In my code, I am also returning a weights table( different from model weights) indicating the importance of each pixel during loss computation; moreover I am returning the input image name for debugging and logging reasons.

**Comment by davidtvs** [comment]
hmm, I think handling loss weights could be something that we could handle in the future is there's a good idea on how to do it. 
What I can do short term is implement your suggestion from the first post so that no error is thrown; however, the content of `*rest` will not be used

**Comment by davidtvs** [comment]
This has been implemented for python 2 and 3 in [v0.1.5](https://pypi.org/project/torch-lr-finder/0.1.5/)

**Comment by vpapaioannou** [comment]
Thanks!
